google-cloud-datastore/src/main/java/com/google/cloud/datastore/testing/LocalDatastoreHelper.java

```diff
@@ -113,11 +115,95 @@ private LocalDatastoreHelper(double consistency, int port) {
     emulatorRunners = ImmutableList.of(gcloudRunner, downloadRunner);
   }
 
+  /** A builder for {@code LocalDatastoreHelper} objects. */
+  public static class Builder {
+    private List<EmulatorRunner> emulatorRunners;
+    private double consistency;
+    private int port;
+    private Path dataDir;
+    private boolean storeOnDisk = Boolean.TRUE.booleanValue();
+
+    private Builder() {}
+
+    public Builder setConsistency(double consistency) {
+      this.consistency = consistency;
+      return this;
+    }
+
+    public Builder setPort(int port) {
+      this.port = port;
+      return this;
+    }
+
+    public Builder setDataDir(Path dataDir) {
+      this.dataDir = dataDir;
+      return this;
+    }
+
+    public Builder setStoreOnDisk(boolean storeOnDisk) {
+      this.storeOnDisk = storeOnDisk;
+      return this;
+    }
+
+    /** Creates a {@code LocalDatastoreHelper} object. */
+    public LocalDatastoreHelper build() {
+      return new LocalDatastoreHelper(this);
+    }
+  }
+
+  LocalDatastoreHelper(Builder builder) {
+    super(
+        "datastore",
+        builder.port > 0 ? builder.port : BaseEmulatorHelper.findAvailablePort(DEFAULT_PORT),
+        PROJECT_ID_PREFIX + UUID.randomUUID().toString());
```

Comment:
Instead of port check here, Can we not do it inside `build()`? assuming the only place it is being called is `LocalDatastoreHelper#build`.

Or better way would be to throw exception if is negative from build() itself

Reply:
yes we can but this looks more often instead of assign it in `build()`